projects/i18n/switch/language-switcher.ts

```diff
@@ -0,0 +1,53 @@
+import {Inject, Injectable, Optional} from '@angular/core';
+import {TuiLanguage, TuiLanguageName} from '@taiga-ui/i18n/interfaces';
+import {TUI_DEFAULT_LANGUAGE} from '@taiga-ui/i18n/tools';
+import {BehaviorSubject, Observable, of} from 'rxjs';
+
+import {TUI_LANGUAGE_CHUNK_LOADER} from './language-chunk-loader-token';
+import {tuiAsyncLoadLanguage} from './load-lang';
+import {
+    TUI_LANGUAGE_DEFAULT_STORAGE,
+    TUI_LANGUAGE_STORAGE_KEY,
+    TuiLangChunkLoader,
+} from './options';
+
+// @dynamic
+@Injectable({providedIn: `root`})
+export class TuiLanguageSwitcher extends BehaviorSubject<Observable<TuiLanguage>> {
+    constructor(
+        @Inject(TUI_DEFAULT_LANGUAGE)
+        private readonly fallback: TuiLanguage,
+        @Inject(TUI_LANGUAGE_STORAGE_KEY)
+        private readonly key: string,
+        @Inject(TUI_LANGUAGE_DEFAULT_STORAGE)
+        private readonly storage: Storage,
+        @Optional()
+        @Inject(TUI_LANGUAGE_CHUNK_LOADER)
+        private readonly loader: TuiLangChunkLoader | null,
+    ) {
+        super(tuiAsyncLoadLanguage({key, storage, fallback, loader}));
+    }
+
+    get currentLanguage(): TuiLanguageName {
```

Comment:
It is `BehaviorSubject`.
Let's rename this getter to `value` to correctly implements [its interface](https://github.com/ReactiveX/rxjs/blob/9a85e01625a1ee7a23e4ab2c2c6d3089a58e2e54/src/internal/BehaviorSubject.ts#L16) ?

Reply:
<img width="717" alt="image" src="https://user-images.githubusercontent.com/12021443/182602131-f52e0c8f-e0fc-4748-b179-686b24bbcdd8.png">